lib/string_reverse.rb

```diff
@@ -1,6 +1,17 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)

```

Comment:
Correct, but why?

Here's how I answered these. This is about as thorough as I can think to be in analyzing space & time complexity: https://github.com/mmcknett/string_reverse/blob/implementation/lib/string_reverse.rb#L3-L11